docs/gen-ai/mcp.md

```diff
@@ -0,0 +1,727 @@
+| [`network.protocol.version`](/docs/attributes-registry/network.md) | string | The version of JSON RPC protocol used. | `1.1`; `2` | `Recommended` when it's not `2.0`. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`network.transport`](/docs/attributes-registry/network.md) | string | The transport protocol used for the MCP session. [4] | `tcp`; `udp` | `Recommended` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`server.address`](/docs/attributes-registry/server.md) | string | Server domain name if available without reverse DNS lookup; otherwise, IP address or Unix domain socket name. [5] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | `Recommended` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`server.port`](/docs/attributes-registry/server.md) | int | Server port number. [6] | `80`; `8080`; `443` | `Recommended` When `server.address` is set | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`mcp.request.argument.<key>`](/docs/attributes-registry/mcp.md) | string | Additional arguments passed to the request within `params` object. `<key>` being the normalized argument name name (lowercase), the value being the argument value. [7] | `mcp.request.argument.location="Seattle, WA"`; `mcp.request.argument.a="42"` | `Opt-In` | ![Development](https://img.shields.io/badge/-development-blue) |
+
+**[1] `error.type`:** This attribute SHOULD be set to the string representation of the JSON RPC
+error code, if one is returned.
+
+When JSON RPC call is successful, but an error is returned within the
+result payload, this attribute SHOULD be set to the low-cardinality
+string representation of the error. When
+[CallToolResult](https://github.com/modelcontextprotocol/modelcontextprotocol/blob/d59fb7e3c1936730fbc93f6ef47790151b2e318a/schema/2025-03-26/schema.ts#L698)
+is returned with `isError` set to `true`, this attribute SHOULD be set to
+`tool_error`.
+
+**[2] `mcp.resource.uri`:** This is a URI of the resource provided in the following requests or notifications: `resources/read`, `resources/subscribe`, `resources/unsubscribe`, or `notifications/resources/updated`.
+
+**[3] `mcp.resource.uri`:** When the client executes a request type that includes a resource URI parameter.
+
+**[4] `network.transport`:** This attribute SHOULD be set to `tcp` or `quic` if the transport protocol
+is HTTP.
+It SHOULD be set to `pipe` if the transport is stdio.
+
+**[5] `server.address`:** When observed from the client side, and when communicating through an intermediary, `server.address` SHOULD represent the server address behind any intermediaries, for example proxies, if it's available.
+
+**[6] `server.port`:** When observed from the client side, and when communicating through an intermediary, `server.port` SHOULD represent the server port behind any intermediaries, for example proxies, if it's available.
```

Comment:
These server attributes are already elsewhere in the semantic conventions. Should those be cross-referenced?

Reply:
they are cross-referenced - there is a link in the table. We don't add links on every mention, but it's a great point - we could have.